list-ops/example.lua

```diff
@@ -0,0 +1,26 @@
+local function reduce(xs, value, f)
+  for _, x in ipairs(xs) do
+    value = f(x, value)
+  end
+  return value
+end
+
+local function map(xs, f)
+  return reduce(xs, {}, function(x, acc)
+    table.insert(acc, f(x))
+    return acc
+  end)
+end
+
+local function filter(xs, pred)
+  return reduce(xs, {}, function(x, acc)
+    if pred(x) then table.insert(acc, x) end
+    return acc
+  end)
+end
+
```

Comment:
It is super interesting, I was drawn, feels like algebra. I tried to compare to [underscorejs](http://underscorejs.org/docs/underscore.html) implementation, but they don't do it that way.